pkg/app/piped/analysisprovider/metrics/datadog/datadog.go

```diff
@@ -97,11 +99,12 @@ func (p *Provider) Type() string {
 }
 
 // RunQuery issues an HTTP request to the API named "MetricsApi.QueryMetrics", then evaluate its response.
-// It performs the given query for the period specified by its own queried time period using the current
-// time as the end of the queried time period.
-//
 // See more: https://docs.datadoghq.com/api/latest/metrics/#query-timeseries-points
-func (p *Provider) RunQuery(ctx context.Context, query string, expected config.AnalysisExpected) (bool, error) {
+func (p *Provider) RunQuery(ctx context.Context, query string, evaluator metrics.Evaluator, queryRange metrics.QueryRange) (bool, error) {
```

Comment:
nit: `queryRange` -> `range` and let's move `evaluator` to be the last parameter.

Reply:
I feel you, but unfortunately `range` is a reserved word in Golang.

> let's move evaluator to be the last parameter.

👍